http-client/src/main/java/com/facebook/airlift/http/client/thrift/ThriftProtocolUtils.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.airlift.http.client.thrift;
+
+import com.facebook.drift.codec.ThriftCodec;
+import com.facebook.drift.protocol.TBinaryProtocol;
+import com.facebook.drift.protocol.TCompactProtocol;
+import com.facebook.drift.protocol.TFacebookCompactProtocol;
+import com.facebook.drift.protocol.TProtocol;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+
+public class ThriftProtocolUtils
+{
+    private ThriftProtocolUtils()
+    {
+    }
+
+    public static <T> void write(T instance, ThriftCodec<T> thriftCodec, ThriftProtocol thriftProtocol, OutputStream out)
+            throws Exception
```

Comment:
Can we use a more specific exception here (and in other methods)? e.g.: `IOException`/ `ProtocolException`?